superset/utils/pandas_postprocessing.py

```diff
@@ -517,3 +518,29 @@ def _parse_location(location: str) -> Tuple[float, float, float]:
         return _append_columns(df, geodetic_df, columns)
     except ValueError:
         raise QueryObjectValidationError(_("Invalid geodetic string"))
+
+
+def contribution(
+    df: DataFrame, orientation: PostProcessingContributionOrientation
+) -> DataFrame:
```

Comment:
Maybe it makes more things to keep things simple, but is it possible to allow specifying which column(s) to compute on? I could imagine this operator being used to support "percent metrics" in table charts and it’ll need the ability to select specific columns.

Maybe the API can be something like this

```
{
  # compute row wise contribution and override the columns
  "columns": ["foo", "bar"],
  "orientation": "row"
}

{
  # compute column wise computation and insert new columns
  "columns": ["pct_abc"],
  "sources": ["abc"]
}
```

Reply:
The thought did occur to me. In this iteration I wanted to add functionality to get feature parity with `viz.py`, but I look forward to adding this optionality later. One significant feature that's currently missing is being able to get feedback from the operation on what happened, e.g. for pivots getting a list of all newly created columns. I have a basic idea of what this could look like, but didn't want to get into that before I actually need it.

FYI, this feature was added to create a line chart with similar features as the legacy one on `viz.py`. In this case I'm chaining two operations together: first a pivot, then a contribution if that option is selected. As I currently don't know what come out of the pivot, I'm unable to tell the contribution operation which columns it should be processing on. Hence I'm just assuming that we're calculating on all columns except the temporal one.